jmix-pivottable/pivottable-flowui/src/main/java/io/jmix/pivottableflowui/export/PivotTableExporterImpl.java

```diff
@@ -36,22 +40,30 @@
  */
 @Component("pvttbl_PivotTableExporter")
 @Scope(BeanDefinition.SCOPE_PROTOTYPE)
-public class PivotTableExporterImpl implements PivotTableExporter {
+public class PivotTableExporterImpl implements PivotTableExporter, ApplicationContextAware, InitializingBean {
 
     public static final Set<Renderer> supportedRenderers = Collections.unmodifiableSet(Sets.newHashSet(
             Renderer.TABLE, Renderer.TABLE_BAR_CHART, Renderer.HEATMAP, Renderer.ROW_HEATMAP, Renderer.COL_HEATMAP));
 
+    protected ApplicationContext applicationContext;
     protected PivotTableExcelExporter excelExporter;
     protected PivotTable<?> pivotTable;
     protected String fileName;
 
-    public PivotTableExporterImpl(PivotTable<?> pivotTable, PivotTableExcelExporter exporter) {
+    public PivotTableExporterImpl(PivotTable<?> pivotTable) {
         Preconditions.checkNotNullArgument(pivotTable);
-        Preconditions.checkNotNullArgument(exporter);
 
         this.pivotTable = pivotTable;
-        this.excelExporter = exporter;
-        this.excelExporter.init(pivotTable);
+    }
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public void afterPropertiesSet() {
+        this.excelExporter = applicationContext.getBean(PivotTableExcelExporter.class, pivotTable);
```

Comment:
According to the comment above, we should allow the exporter to be defined externally.
So here we should add a condition:
```java
if (this.excelExporter == null) {
    this.excelExporter = applicationContext.getBean(....);
}
```

Reply:
Also add a comment that this `condition` should be removed with **Release 3.0**.